src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -49,10 +54,41 @@ public partial class SourceArchive // Partial required for serialization
         public const string SourceKind_ArmTemplate = "armTemplate";
         public const string SourceKind_TemplateSpec = "templateSpec";
 
-        private const string MetadataFileName = "__metadata.json";
+        private const string MetadataFileName = "metadata.json";
+        private const string FilesFolderName = "files";
+
+        private static readonly char[] PathCharsToAvoid = Path.GetInvalidFileNameChars()
```

Comment:
[nit] We should stick to readonly or immutable data structures for any shared data.

Reply:
022fe99c2